cloud/logstream.go

```diff
@@ -24,12 +24,14 @@ func (s *StreamError) Error() string {
 	return s.Err.Error()
 }
 
-func (c *Client) StreamLogs(ctx context.Context, man *pb.RunManifest, ch <-chan *pb.Delta) []error {
+func (c *Client) StreamLogs(ctx context.Context, man *pb.RunManifest, ch <-chan *pb.Delta, errCh chan<- error) {
```

Comment:
Just a pattern idea: What if this StreamLogs function was async. It would start off the goroutine within it, and it would construct errCh itself and return it immediately.

The caller can wait for errCh to close to know that the streaming is done.

It would make the API a little cleaner.

Reply:
I like it. I'll take another look in the next week or so. I'll merge this for now as it fixes a flaky test.

Reply:
Actually, it was a pretty straightforward change. Updated!